### PR TITLE
feat: [asset-swapper] Shell  🐚

### PR DESCRIPTION
**Pull request by dekz**
## Description

```
SELL USDT->USDC 542.74 -> 542.60 ($542.74) after 16.3s @ prod
	✔ PASS 542.74 -> 542.60 ($535.20) ($7.40)
	Swerve: 100%
	gas: 321990

SELL USDT->USDC 542.74 -> 542.55 ($542.74) after 16.3s @ dev
	✔ PASS 542.74 -> 542.55 ($532.71) ($9.83)
	Shell: 100%
	gas: 428071
```

```
BUY DAI->USDC 24940.39 -> 25074.17 ($25073.14) after 0.7s @ prod
	✔ PASS 24964.33 -> 25098.18 ($25237.21) ($23.24)
	Uniswap_V2: 34.8%, Balancer: 11.29%, Mooniswap: 53.910000000000004%
	gas: 802899
BUY DAI->USDC 24845.60 -> 25074.17 ($25073.14) after 0.7s @ dev
	✔ PASS 25094.06 -> 25323.15 ($25358.85) ($13.84)
	Shell: 100%
	gas: 421806
BUY USDC->USDT 110054.07 -> 109988.04 ($109988.04) after 15.3s @ prod
	✔ PASS 111154.62 -> 111087.92 ($111174.10) ($24.04)
	mStable: 100%
	gas: 830843
BUY USDC->USDT 110026.54 -> 109988.04 ($109988.04) after 15.3s @ dev
	✔ PASS 111126.80 -> 111087.91 ($111134.63) ($12.39)
	Shell: 100%
	gas: 438663
BUY USDC->DAI 5413.41 -> 5306.84 ($5359.90) after 4.7s @ dev
	✔ PASS 5467.54 -> 5359.82 ($5481.92) ($14.60)
	Shell: 100%
	gas: 444766
BUY USDC->DAI 5372.94 -> 5306.84 ($5359.90) after 4.7s @ prod
	✔ PASS 5426.66 -> 5359.89 ($5440.61) ($14.17)
	MultiHop: 100%
	gas: 495590
BUY DAI->USDT 48927.95 -> 49022.73 ($49022.73) after 13.1s @ prod
	✔ PASS 49417.23 -> 49512.25 ($49960.75) ($49.34)
	MultiHop: 100%
	gas: 1726045
BUY DAI->USDT 48795.29 -> 49022.73 ($49022.73) after 13.1s @ dev
	✔ PASS 49283.24 -> 49510.42 ($49790.40) ($14.32)
	Shell: 100%
	gas: 436280
```

## Checklist:

-   [ ] Prefix PR title with `[WIP]` if necessary.
-   [ ] Add tests to cover changes as needed.
-   [ ] Update documentation as needed.
-   [ ] Add new entries to the relevant CHANGELOG.jsons.

### CODE REVIEW

File: packages/asset-swapper/contracts/src/ShellSampler.sol [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+        takerTokenAmounts = new uint256[](numSamples);
+
+        for (uint256 i = 0; i < numSamples; i++) {
+            (bool didSucceed, bytes memory resultData) =
+                address(_getShellAddress()).staticcall.gas(DEFAULT_CALL_GAS)(
+                    abi.encodeWithSelector(
+                        IShell(0).viewTargetSwap.selector,
+                        takerToken,
+                        makerToken,
+                        takerTokenAmounts[i]
```
Comment (dorothy-zbornak) [inline packages/asset-swapper/contracts/src/ShellSampler.sol]:
```suggestion
                        makerTokenAmounts[i]
```
?

File: contracts/asset-proxy/contracts/src/bridges/ShellBridge.sol [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+import "../interfaces/IShell.sol";
+
+
+contract ShellBridge is
+    IERC20Bridge,
+    IWallet,
+    DeploymentConstants
+{
+
+    /// @dev Swaps specified tokens against the mStable mUSD contract
```
Comment (dorothy-zbornak) [inline contracts/asset-proxy/contracts/src/bridges/ShellBridge.sol]:
```suggestion
    /// @dev Swaps specified tokens against the shell contract
```

File: contracts/asset-proxy/contracts/src/bridges/ShellBridge.sol [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    /// @param toTokenAddress The token to give to `to` (i.e DAI, USDC, USDT).
```
Comment (dorothy-zbornak) [inline contracts/asset-proxy/contracts/src/bridges/ShellBridge.sol]:
```suggestion
    /// @param toTokenAddress The token to give to `to`.
```

**Review by dorothy-zbornak — APPROVED** [review]
Simbot buys smoke test?

File: packages/asset-swapper/contracts/src/ShellSampler.sol (hunk 1 above)
Comment (dekz) [reply packages/asset-swapper/contracts/src/ShellSampler.sol]: ain't ran the buys yet!!!!

**Pull request merged**